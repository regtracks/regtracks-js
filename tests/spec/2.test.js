const Track = require('../../index');

describe('tests section 2 of the specification', () => {
  test('sections 2.2-3: rule order', () => {
    let schema = String.raw`
    @foobar
    ||
    (a)
    (b)
    any c, d, e
    ||
    `;

    let text1 = 'abd';
    let text2 = 'abf';
    let text3 = 'abfabe';

    const track = new Track(schema);

    expect(track.test(text1, 'foobar')).toBe(true);
    expect(track.test(text2, 'foobar')).toBe(false);
    expect(track.test(text3, 'foobar')).toBe(true);
  });

  test('section 2.4: pattern delimiters', () => {
    let schema = String.raw`
    @named_pattern
    ||
    (test?)
    none a to z
    ||
    `;

    const track = new Track(schema);
    expect(track.parser.tree.tree).toMatchSnapshot();
  });

  test('section 2.5: empty lines', () => {
    let schema = String.raw`
      @pattern
      ||

      (a)

      any digit

      ||


    `;

    let schema2 = String.raw`
      @pattern
      ||
      (a)
      any digit
      ||
    `;

    const track = new Track(schema);
    const track2 = new Track(schema2);

    expect(track.parser.tree.tree).toEqual(track2.parser.tree.tree);
  });

  test('section 2.6: pattern without symbol', () => {
    let schema = String.raw`
      ||
      (a)
      (b)
      any digit
      ||
    `;

    let text = 'ab5';

    const track = new Track(schema);
    expect(track.test(text)).toBe(true);
  });
});
