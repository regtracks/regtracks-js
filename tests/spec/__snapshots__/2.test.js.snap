// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests section 2 of the specification section 2.4: pattern delimiters 1`] = `
Object {
  "named_pattern": Object {
    "name": "named_pattern",
    "root": Object {
      "children": Array [
        Object {
          "collectionIdent": undefined,
          "forks": false,
          "inclusive": true,
          "matches": Array [
            Object {
              "matches": "test?",
              "type": "literal_criterion",
            },
          ],
          "optional": false,
          "repeat": undefined,
          "type": "matching_rule",
        },
        Object {
          "collectionIdent": undefined,
          "forks": false,
          "inclusive": false,
          "matches": Array [
            Object {
              "matches": /\\[a-z\\]/,
              "type": "character_range_criterion",
            },
          ],
          "optional": false,
          "repeat": undefined,
          "type": "matching_rule",
        },
      ],
      "collectionIdent": undefined,
      "forks": false,
      "optional": false,
      "parent": undefined,
      "repeat": undefined,
      "type": "block",
    },
    "type": "pattern",
  },
}
`;
